clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java

```diff
@@ -244,6 +273,26 @@ public CompletableFuture start() throws Exception {
                     hubConnectionState = HubConnectionState.CONNECTED;
                     connectionState = new ConnectionState(this);
                     logger.log(LogLevel.Information, "HubConnected started.");
+
+                    resetServerTimeout();
+                    this.pingTimer = new Timer();
+                    this.pingTimer.schedule(new TimerTask() {
+                        @Override
+                        public void run() {
+                            try {
+                                if (System.currentTimeMillis() > nextServerTimeout) {
+                                    stop("Server timeout elapsed without receiving a message from the server.");
+                                    return;
+                                }
+
+                                if (System.currentTimeMillis() > nextPingActivation) {
+                                    sendHubMessage(PingMessage.getInstance());
```

Comment:
I assume sendHubMessage is doing the appropriate locking?

Reply:
What locking is that?

Reply:
I'm asking. There's no chance of overlapping sends right?

Reply:
So the queue in the websocket library is locked and the library itself is thread safe?

Reply:
I have confirmed (by reviewing the library code) that the websocket library is thread safe (at least for sends).

See https://github.com/square/okhttp/blob/ceb42f132220c71c7ad9a2948b24ee4f9e7df8e0/okhttp/src/main/java/okhttp3/internal/ws/RealWebSocket.java#L383